common/spring-cloud-stream-modules-test-support/src/main/java/org/springframework/cloud/stream/modules/test/file/remote/TestSftpServer.java

```diff
@@ -38,20 +41,73 @@
 
 	private final SshServer server = SshServer.setUpDefaultServer();
 
-	private final int port;
-
 	public TestSftpServer(String root, int port) {
 		super(root);
-		this.port = port;		
+		setFtpTemporaryFolder(
+				new TemporaryFolder() {
```

Comment:
Hmmm, yeah, seems like missuse of `TemporaryFolder`.
Also, forces some paths in `TestSftpServer` which makes it less re-useable. Is there a better way?